package/src/main/java/com/azure/cosmos/cassandra/CosmosRetryPolicy.java

```diff
@@ -0,0 +1,301 @@
+    }
+
+    // endregion
+
+    // region Accessors
+
+    CosmosRetryPolicy(
+        final int maxRetryCount, final int fixedBackOffTimeInMillis, final int growingBackoffTimeInMillis) {
+
+        this.maxRetryCount = maxRetryCount;
+        this.fixedBackoffTimeInMillis = fixedBackOffTimeInMillis;
+        this.growingBackoffTimeInMillis = growingBackoffTimeInMillis;
+    }
+
+    /**
+     * Gets the {@code fixed-backoff-time} specified by this {@link CosmosRetryPolicy} object.
+     *
+     * @return the {@code fixed-backoff-time} specified by this {@link CosmosRetryPolicy} object.
+     */
+    public int getFixedBackoffTimeInMillis() {
+        return this.fixedBackoffTimeInMillis;
+    }
+
+    /**
+     * Gets the {@code growing-backoff-time} specified by this {@link CosmosRetryPolicy} object.
+     *
+     * @return the {@code growing-backoff-time} specified by this {@link CosmosRetryPolicy} object.
+     */
+    public int getGrowingBackoffTimeInMillis() {
+        return this.growingBackoffTimeInMillis;
+    }
+
+    /**
+     * Gets the {@code max-retries} specified by this {@link CosmosRetryPolicy} object.
+     *
+     * @return the {@code max-retries} specified by this {@link CosmosRetryPolicy} object.
+     */
+    public int getMaxRetryCount() {
+        return this.maxRetryCount;
+    }
+
+    // endregion
+
+    // region Methods
+
+    /**
+     * Closes the current {@link CosmosRetryPolicy}.
+     */
+    @Override
+    public void close() {
+        // nothing to do
+    }
+
+    @Override
+    public RetryDecision onErrorResponse(
+        @NonNull final Request request, @NonNull final CoordinatorException error, final int retryCount) {
+
+        RetryDecision retryDecision;
+
+        try {
+            if (error instanceof OverloadedException || error instanceof WriteFailureException) {
+                if (this.maxRetryCount == -1 || retryCount < this.maxRetryCount) {
+                    int retryAfterMillis = getRetryAfterMillis(error.toString());
+                    if (retryAfterMillis == -1) {
+                        retryAfterMillis = this.maxRetryCount == -1
+                            ? this.fixedBackoffTimeInMillis
+                            : this.growingBackoffTimeInMillis * retryCount + RANDOM.nextInt(
+                                GROWING_BACKOFF_SALT_IN_MILLIS);
+                    }
+                    Thread.sleep(retryAfterMillis);
+                    retryDecision = RetryDecision.RETRY_SAME;
+                } else {
+                    retryDecision = RetryDecision.RETHROW;
+                }
+            } else {
+                retryDecision = RetryDecision.RETHROW;
+            }
+
+        } catch (final InterruptedException exception) {
+            retryDecision = RetryDecision.RETHROW;
+        }
+
+        return retryDecision;
+    }
+
+    @Override
+    public RetryDecision onReadTimeout(
```

Comment:
Can we follow the same logic that the DefaultRetryPolicy follows for Read and Write Timeouts?

Reply:
We've added some handling on the server side for this: https://msdata.visualstudio.com/CosmosDB/_git/CosmosDB?path=%2FProduct%2FCosmos%2FCassandra%2FCore%2FResponses%2FErrorResponse.cs&version=GBmaster&line=520&lineEnd=521&lineStartColumn=1&lineEndColumn=1&lineStyle=plain&_a=contents

It could possibly be revisited and updated though.